### PR TITLE
feat(csv): parser implementation

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
Signed-off-by: Vincenzo Palazzo <vincenzopalazzodev@gmail.com>

### CODE REVIEW

File: frontend_csv/src/parser/parser.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+/// Core implementation of the parser
+
+pub struct Parser {}
+
+impl Parser {
+    pub fn new() -> Parser {
+        return Parser {};
+    }
+
+    pub fn parse_token(&mut self, source: &Vec<CSVToken>) -> Vec<(AstMsgLineType, Vec<String>)> {
```
Comment (vincenzopalazzo) [inline frontend_csv/src/parser/parser.rs:18]: Why do you return a vector of `Vec<(AstMsgLineType, Vec<String>)>`? What is the String vector?

File: frontend_csv/src/parser/ast_msg.rs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+}
+
+pub trait AstMsgTrait {
+    fn get_msg_type(&self) -> AstMsgType;
+    fn get_values(&self) -> String;
+    fn check_valid(&mut self) -> bool;
+}
+
+impl InitMsg {
+    pub fn new(lines: Vec<(AstMsgLineType, Vec<String>)>) -> InitMsg {
```
Comment (vincenzopalazzo) [inline frontend_csv/src/parser/ast_msg.rs:50]:
```suggestion
    pub fn new(lines: &Vec<(AstMsgLineType, Vec<String>)>) -> InitMsg {
```

**Review by vincenzopalazzo — COMMENTED** [review]
@michiboo The rust code need to be improved, there are too much clone and copy vector in this code!

In addition, I do not like the solution because it is confusing and I think it is not extendible when I will add the support for other bolt12

The parser entry point need to looks like the klox entry point
```
    fun parse(): List<IStatement> {
        val statements = ArrayList<IStatement>()
        while (!isAtEnd()) {
            val stmt = declaration()
            statements.add(stmt!!)
        }
        return statements
    }
```

Where the `IStatement` in your case is `AstMsgTrait`

File: frontend_csv/src/parser/parser.rs (hunk 1 above)
Comment (michiboo) [reply frontend_csv/src/parser/parser.rs:18]: the string vector is the value, for example ["msgtype", "Init", 16] is the value, while the line type is AstMsgLineType:msgtype... need those value to make check if the interger value is 16 or store the "init" for future lines reference checking

**Comment by vincenzopalazzo** [comment]
Closing this because the git history is fucked! You reopen it when you finish the implementation

**Pull request closed without merging**